src/lib/rnp.c

```diff
@@ -1252,11 +1250,12 @@ rnp_generate_key(rnp_t *rnp, char *id, int numbits)
     pgp_key_t *    key;
     pgp_io_t *     io;
     uint8_t *      uid;
-    char           passphrase[128];
-    char           newid[1024];
-    char           filename[MAXPATHLEN];
-    char           dir[MAXPATHLEN];
-    char *         cp;
+    char           passphrase[128] = { 0 };
+    char           newid[1024] = { 0 };
+    char           filename[MAXPATHLEN] = { 0 };
+    char           dir[MAXPATHLEN] = { 0 };
+    char           keyid[2*PGP_KEY_ID_SIZE + 1] = { 0 };
```

Comment:
No that constant is (AIUI) for user ids - this var is holding an 8-byte long hex keyid